File: 2024/day11/python/test_problem.py
import unittest
from problem import problem_part1, problem_part2

class TestProblem(unittest.TestCase):
    def test_problem_part1_test1(self):
        input_list = [
            "0 1 10 99 999"
        ]
        result = problem_part1(input_list, 1)
        self.assertEqual(result, 7)

    def test_problem_part1_test2(self):
        input_list = [
            "125 17"
        ]
        result = problem_part1(input_list, 6)
        self.assertEqual(result, 22)

    def test_problem_part1_test3(self):
        input_list = [
            "125 17"
        ]
        result = problem_part1(input_list, 25)
        self.assertEqual(result, 55312)

    def test_problem_part2_test1(self):
        input_list = [
            "0 1 10 99 999"
        ]
        result = problem_part2(input_list, 1)
        self.assertEqual(result, 7)

    def test_problem_part2_test2(self):
        input_list = [
            "125 17"
        ]
        result = problem_part2(input_list, 6)
        self.assertEqual(result, 22)

    def test_problem_part2_test3(self):
        input_list = [
            "125 17"
        ]
        result = problem_part2(input_list, 25)
        self.assertEqual(result, 55312)

File: 2024/day2/julia/TestProblem.jl
using Test

include("Problem.jl")

@testset "Problem" begin
    input_vector = [
        "7 6 4 2 1",
        "1 2 7 8 9",
        "9 7 6 2 1",
        "1 3 2 4 5",
        "8 6 4 4 1",
        "1 3 6 7 9"
    ]
    @test problem_part1(input_vector) == 2

    @test problem_part2(input_vector) == 4
end